### ISSUE TITLE
Do Sass partials (_variables.scss) work?

### ISSUE THREAD

**Issue by JosephStocks** [Wscats/compile-hero#38]
I created a Sass partial file called '_variables.scss' that holds the variables.

In my main scss file at the top I use 

    @import "variables";

When I save the _variables.scss file, Compile Hero creates a new css file called "_variables.css". When I save the main css file, I get this error:

     File to import not found or unreadable: variables.
     Parent style sheet: stdin

It should transpile the variables within the main. It should result in one main file.

I thought the underscore partial syntax was general to all sass compilers. Is it not?

**Comment by summers173** [comment]
I was about to report the same, this is a great extension, very useful for testing and demo purposes, to include the ability to compile SCSS partials will be a huge improvement.

Keep the good work and hope this can be addressed, regards.

**Comment by Wscats** [comment]
@JosephStocks @summers173 I have fixed this problem in the latest version, you can update and use it normally.

**Comment by summers173** [comment]
Thanks @Wscats, will give it try, have a nice one

**Comment by norixx** [comment]
@Wscats You meant the latest ver.2.3.4? There are two versions of Compile Hero extension Pro and Not Pro in market place.
Pro version still compiles my partial scss files( Not Pro version works fine :) )

**Comment by Wscats** [comment]
> @Wscats You meant the latest ver.2.3.4? There are two versions of Compile Hero extension Pro and Not Pro in market place.
> Pro version still compiles my partial scss files( Not Pro version works fine :) )

Not Pro(6.8.65) and Pro(2.3.4). Both versions fix this problem, you can try to update both.

**Comment by norixx** [comment]
@Wscats Ok, let me rephrase, problem still do exists. Take a look at attached image.
[imgur](https://imgur.com/bddGTW0)

**Comment by Wscats** [comment]
> @Wscats Ok, let me rephrase, problem still do exists. Take a look at attached image.
> [imgur](https://imgur.com/bddGTW0)

Thank you so much for the feedback in the video.
In your video, there is no problem with your code, but why does it generate an empty file because of this:

```scss
$test: red;
@mixin test {
    font-weight: normal;
    font-size: 24px;
    color: $test;
}
```

> Mixins are defined using the @mixin at-rule, which is written @mixin <name> { ... } or @mixin name(<arguments...>) { ... }. A mixin’s name can be any Sass identifier, and it can contain any statement other than top-level statements. They can be used to encapsulate styles that can be dropped into a single style rule; they can contain style rules of their own that can be nested in other rules or included at the top level of the stylesheet; or they can just serve to modify variables.
> 
> Mixins are included into the current context using the @include at-rule, which is written @include <name> or @include <name>(<arguments...>), with the name of the mixin being included.
[https://sass-lang.com/documentation/at-rules/mixin](https://sass-lang.com/documentation/at-rules/mixin)


If there is only @mixin a code block, sass cannot generate a meaningful style code snippet. So you can try to rewrite it like this.

```scss
$test: red;
@mixin test {
    font-weight: normal;
    font-size: 24px;
    color: $test;
}
.test {
    @include test;
}
```

The above code will be compiled normally into the following code.

```scss
.test {
  font-weight: normal;
  font-size: 24px;
  color: red;
}
```

Hope it helps you, thank you for your use😁

**Comment by norixx** [comment]
I am sorry i was short of my words. I `@imported` _test.scss file into the main scss file (style.scss)  and did used `@include` directive. style.scss was processed fine 👍 
I just did not want my _test.scss to become test.css. That was all.

**Comment by Wscats** [comment]
> I am sorry i was short of my words. I `@imported` _test.scss file into the main scss file (style.scss) and did used `@include` directive. style.scss was processed fine 👍
> I just did not want my _test.scss to become _test.css (I guess it shouldn't). That was all.

If you think the plugin can help you solve the problem, please give us a five-star praise. https://marketplace.visualstudio.com/items?itemName=Wscats.eno&ssr=false#review-details

**Comment by jorgefuertes** [comment]
Same bug with `_*.sass` files.
